src/Ogooreck.Sample.BusinessLogic.FSharp.Tests/Deciders/BankAccount.fs

```diff
@@ -0,0 +1,69 @@
+﻿module Deciders.BankAccount
+
+open System
+open BankAccountPrimitives
+
+type BankAccountOpened =
+    { BankAccountId: AccountId
+      AccountNumber: AccountNumber
+      ClientId: ClientId
+      CurrencyISOCode: CurrencyCode
+      CreatedAt: DateTimeOffset
+      Version: int64 }
+
+type DepositRecorded =
+    { BankAccountId: AccountId
+      Amount: decimal
+      CashierId: CashierId
+      RecordedAt: DateTimeOffset
+      Version: int64 }
+
+type CashWithdrawnFromATM =
+    { BankAccountId: AccountId
+      Amount: decimal
+      ATMId: AtmId
+      RecordedAt: DateTimeOffset
+      Version: int64 }
+
+type BankAccountClosed =
+    { BankAccountId: AccountId
+      Reason: string
+      ClosedAt: DateTimeOffset
+      Version: int64 }
+
+type Event =
+    | BankAccountOpened of BankAccountOpened
+    | DepositRecorded of DepositRecorded
+    | CashWithdrawnFromATM of CashWithdrawnFromATM
+    | BankAccountClosed of BankAccountClosed
+
+type BankAccount =
+    | NotInitialised
+    | Open of
+        {| Id: AccountId
+           Balance: decimal
+           Version: int64 |}
+    | Closed of {| Id: AccountId; Version: int64 |}
+
+let evolve (state: BankAccount) (event: Event) : BankAccount =
+    match (state, event) with
```

Comment:
if you are constructing a tuple to return a result, adding the explicit parens is sometimes a way to make that clear. Within an expresssion like this you never put the parens
```suggestion
    match state, event with
```

Reply:
Thanks for the explanation; updated in https://github.com/oskardudycz/Ogooreck/pull/13/commits/9c3bd4fa7efd66b75bf111513ba31eea4892e0cb.

p.s. I still don't fully grasp where to put parens and where not, I tried to google that but haven't found decent resources on that topic. Same for where to use which type definition syntax (tuples vs union types vs aliases).

Reply:
This guide is very good https://learn.microsoft.com/en-us/dotnet/fsharp/style-guide/formatting#formatting-tuple-expressions

I only put in the parens where my tuple construction is some long expression and I want to be clear about it. (and if rider offers to remove it, I remove it - their default settings are in line with the above reference and me and others whine if its not)

Reply:
Thanks @bartelink for explanation :+1: